CharlieBackend.Api.UnitTes/RepositoriesTests/DashboardRepositoryTest.cs

```diff
@@ -0,0 +1,744 @@
+                    LessonDate = new DateTime(),
+                    StudentMark = 100,
+                },
+                new StudentMarkDto
+                {
+                    CourseId = 2,
+                    StudentGroupId = 1,
+                    StudentId = 12,
+                    LessonId = 10,
+                    LessonDate = new DateTime(),
+                    StudentMark = 30,
+                },
+                new StudentMarkDto
+                {
+                    CourseId = 2,
+                    StudentGroupId = 1,
+                    StudentId = 12,
+                    LessonId = 10,
+                    LessonDate = new DateTime(),
+                    StudentMark = 90,
+                }
+            };
+
+            var mock = GetMockDbSetOfVisits();
+
+            _applicationContextMock.SetupGet(x => x.Visits).Returns(mock.Object);
+
+            var dashboardRepository = new DashboardRepository(_applicationContextMock.Object);
+
+            //Act
+
+            var result = await dashboardRepository.GetStudentsMarksListByStudentIds(new List<long> { 10, 12 });
+
+            //Assert
+
+            Assert.NotNull(result);
+            Assert.NotEmpty(result);
+            Assert.Equal(expected.Count, result.Count);
+            for (int i = 0; i < expected.Count; i++)
+            {
+                Assert.Equal(expected[i].ToString(), result[i].ToString());
+            }
+        }
+
+        [Fact]
+        public async Task GetStudentAverageMarksByStudentIdsAndGropsIdsAsync()
+        {
+            //Arrange
+
+            var expected = new List<AverageStudentMarkDto>
+            {
+                new AverageStudentMarkDto
+                {
+                    CourseId =2,
+                    StudentGroupId =1,
+                    StudentId =11,
+                    StudentAverageMark = 78
+                },
+                new AverageStudentMarkDto
+                {
+                    CourseId =2,
+                    StudentGroupId =1,
+                    StudentId =12,
+                    StudentAverageMark = 60
+                }
+            };
+
+            var mock = GetMockDbSetOfVisits();
+
+            _applicationContextMock.Setup(x => x.Visits).Returns(mock.Object);
+
+            var dashboardRepository = new DashboardRepository(_applicationContextMock.Object);
+
+            //Act
+
+            var result = await dashboardRepository.GetStudentAverageMarksByStudentIdsAndGropsIdsAsync(new List<long> { 11, 12 }, new List<long> { 1 });
+
+            //Assert
+
+            Assert.NotNull(result);
+            Assert.NotEmpty(result);
+            Assert.Equal(expected.Count, result.Count);
+            for (int i = 0; i < expected.Count; i++)
+            {
+                Assert.Equal(expected[i].ToString(), result[i].ToString());
+            }
+        }
+
+        [Fact]
+        public async Task GetStudentAverageVisitsPercentageByStudentIdsAsync()
```

Comment:
What will be returned is you specify studentId that doesn't exist?
What will be returned if a student doesn't have visits?
For those cases, we can write separate tests and call it by convention that I've posted to the group.
For example GetStudentAverageVisitsPercentageByStudentIdsAsync_AllDataIsValid_ReturnEverageValue
GetStudentAverageVisitsPercentageByStudentIdsAsync_StudentDoesNotExist_ReturnError
GetStudentAverageVisitsPercentageByStudentIdsAsync_StudentDoesNotHaveVisits_ReturnUserWithZeroValue

I thing, we can merge this PR for now.